File: examples/python/history_candlesticks.py
import datetime

from longbridge.openapi import (
    QuoteContext,
    Config,
    OAuthBuilder,
    Period,
    AdjustType,
    TradeSessions,
)

oauth = OAuthBuilder("your-client-id").build(
    lambda url: print(f"Open this URL to authorize: {url}")
)
config = Config.from_oauth(oauth)
ctx = QuoteContext(config)

# get candlesticks by offset
print("get candlesticks by offset")
print("====================")
candlesticks = ctx.history_candlesticks_by_offset(
    "700.HK",
    Period.Day,
    AdjustType.NoAdjust,
    False,
    10,
    datetime.datetime(2023, 8, 18),
    TradeSessions.Intraday,
)
for candlestick in candlesticks:
    print(candlestick)

# get candlesticks by date
print("get candlesticks by date")
print("====================")
candlesticks = ctx.history_candlesticks_by_date(
    "700.HK",
    Period.Day,
    AdjustType.NoAdjust,
    datetime.date(2022, 5, 5),
    datetime.date(2022, 6, 23),
    TradeSessions.Intraday,
)
for candlestick in candlesticks:
    print(candlestick)

File: examples/nodejs/history_candlesticks.js
const {
  Config,
  QuoteContext,
  Period,
  AdjustType,
  TradeSessions,
  NaiveDate,
  NaiveDatetime,
  Time,
  OAuth,
} = require('longbridge');

async function main() {
  const oauth = await OAuth.build("your-client-id", (_, url) => {
    console.log("Open this URL to authorize: " + url);
  });
  const config = Config.fromOAuth(oauth);
  const ctx = await QuoteContext.new(config);

  // get candlesticks by offset
  console.log("get candlesticks by offset");
  console.log("====================");
  const datetime = new NaiveDatetime(new NaiveDate(2023, 8, 18), new Time(0, 0, 0));
  const byOffset = await ctx.historyCandlesticksByOffset(
    "700.HK",
    Period.Day,
    AdjustType.NoAdjust,
    false,
    datetime,
    10,
    TradeSessions.Intraday,
  );
  for (const candlestick of byOffset) {
    console.log(candlestick.toString());
  }

  // get candlesticks by date
  console.log("get candlesticks by date");
  console.log("====================");
  const byDate = await ctx.historyCandlesticksByDate(
    "700.HK",
    Period.Day,
    AdjustType.NoAdjust,
    new NaiveDate(2022, 5, 5),
    new NaiveDate(2022, 6, 23),
    TradeSessions.Intraday,
  );
  for (const candlestick of byDate) {
    console.log(candlestick.toString());
  }
}

main().catch(console.error);
